File: cs50/pset1/cash/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    int cents = 0, count = 0;

    do
    {
        cents = round(100 * get_float("Enter owed amount in $: "));
    }
    while (cents < 0);

    (cents / 25 != 0) ? count += cents / 25 : 1 * 1;
    cents %= 25;

    (cents / 10 != 0) ? count += cents / 10 : 1 * 1;
    cents %= 10;

    (cents / 5 != 0) ? count += cents / 5 : 1 * 1;
    cents %= 5;

    count += cents;

    printf("%i\n", count);
}

File: cs50/pset6/greedy/cash.py
from cs50 import get_float

# get a positive decimal number as user input
while True:
    cents = round(100 * get_float("Enter owed amount: "))
    if cents > 0:
        break

# this will hold the number of coins
count = 0

# check if we can change using 25 cents coins
if cents >= 25:
    count += cents // 25
    cents %= 25

# check if we can change using 10 cents coins
if cents >= 10:
    count += cents // 10
    cents %= 10

# check if we can change using 5 cents coins
if cents >= 5:
    count += cents // 5
    cents %= 5

# whatever has remained will only be changed using
# 1 cent coins, so we just add that number
count += cents

print(count)